src/deluge/dsp/filter/filter.h

```diff
@@ -54,23 +57,68 @@ class Filter {
 	 * @param sampleIncrement increment between samples
 	 * @param extraSaturation extra saturation value
 	 */
-	void filterMono(q31_t* startSample, q31_t* endSample, int32_t sampleIncrememt = 1) {
-		static_cast<T*>(this)->doFilter(startSample, endSample, sampleIncrememt);
+	[[gnu::hot]] void filterMono(q31_t* startSample, q31_t* endSample, int32_t sampleIncrememt = 1) {
+		if (wetLevel == ONE_Q31) {
+			static_cast<T*>(this)->doFilter(startSample, endSample, sampleIncrememt);
+		}
+		else {
+			memcpy(blendBuffer, startSample, (endSample - startSample) * sizeof(q31_t));
+			static_cast<T*>(this)->doFilter(startSample, endSample, sampleIncrememt);
+
+			q31_t* currentSample = startSample;
+			q31_t* currentDrySample = blendBuffer;
+			do {
+				q31_t wet = multiply_32x32_rshift32(*currentSample, wetLevel);
+				*currentSample = multiply_accumulate_32x32_rshift32_rounded(wet, *currentDrySample, ONE_Q31 - wetLevel)
+				                 << 1;
+				currentSample += 1;
+				currentDrySample += 1;
+				updateBlend(); // will go to one over 512 samples
+			} while (currentSample < endSample);
+		}
 	}
 	/**
 	 * Filter a buffer of interleaved stereo samples from startSample to endSample incrememnting by the increment
 	 * @param startSample pointer to first sample in buffer
 	 * @param endSample pointer to last sample
 	 * @param extraSaturation extra saturation value
 	 */
-	void filterStereo(q31_t* startSample, q31_t* endSample) {
-		static_cast<T*>(this)->doFilterStereo(startSample, endSample);
-		;
+	[[gnu::hot]] void filterStereo(q31_t* startSample, q31_t* endSample) {
+		if (wetLevel == ONE_Q31) {
+			static_cast<T*>(this)->doFilterStereo(startSample, endSample);
+		}
+		else {
+			memcpy(blendBuffer, startSample, (endSample - startSample) * sizeof(q31_t));
```

Comment:
```suggestion
			std::copy(startSample, endSample, &blendBuffer[0]);
```
Double check the binary size but I think it should be fine.

Reply:
we have a faster arm assembly memcpy implementation than libc/c++ and it makes a meaningful difference

It's in c_lib_alternatives.s